libkbfs/prefetcher.go

```diff
@@ -186,27 +217,44 @@ func (p *blockPrefetcher) PrefetchBlock(
 	// TODO: Remove this log line.
 	p.log.CDebugf(context.TODO(), "Prefetching block by request from "+
 		"upstream component. Priority: %d", priority)
-	return p.request(priority, kmd, ptr, block, "")
+	// TODO: pass this channel back up in case the caller wants to be notified
+	// on prefetch completion.
+	var wg sync.WaitGroup
+	wg.Add(1)
+	return p.request(priority, kmd, ptr, block, "", &wg)
 }
 
 // PrefetchAfterBlockRetrieved implements the Prefetcher interface for
-// blockPrefetcher.
+// blockPrefetcher. Returns a channel that is closed once all the prefetches
+// complete.
 func (p *blockPrefetcher) PrefetchAfterBlockRetrieved(
-	b Block, ptr BlockPointer, kmd KeyMetadata) {
+	b Block, ptr BlockPointer, kmd KeyMetadata) <-chan struct{} {
+	doneCh := make(chan struct{})
+	var wg *sync.WaitGroup
 	switch b := b.(type) {
 	case *FileBlock:
 		if b.IsInd {
-			p.prefetchIndirectFileBlock(b, kmd)
+			wg = p.prefetchIndirectFileBlock(b, kmd)
+		} else {
+			close(doneCh)
+			return doneCh
 		}
 	case *DirBlock:
 		if b.IsInd {
-			p.prefetchIndirectDirBlock(b, kmd)
+			wg = p.prefetchIndirectDirBlock(b, kmd)
 		} else {
-			p.prefetchDirectDirBlock(ptr, b, kmd)
+			wg = p.prefetchDirectDirBlock(ptr, b, kmd)
 		}
 	default:
 		// Skipping prefetch for block of unknown type (likely CommonBlock)
+		close(doneCh)
+		return doneCh
 	}
+	go func() {
```

Comment:
Hah I had it that way and changed to the current semantics. I'll change it to match your expectations.